### ISSUE TITLE
File management dös Not work at all for me

### ISSUE THREAD

**Issue by henkelma** [holzschu/Carnets#23]
All previously saved files are greyed out and can‘t be opened.

Save as does not work at all. The dialog box is prepopulated with a not existing path.

**Comment by holzschu** [comment]
Hi,
which version are you using? Did something change recently, or was it working before? 

Also, I'll have to know in which directory are the files you are working on when you do "Save as".

**Comment by henkelma** [comment]
I‘m using 1.2, but the behaviour of 1.0 was the same
![IMG_0252](https://user-images.githubusercontent.com/44764104/57522833-7d046380-7323-11e9-87fb-cb7d3ff7ab1e.jpeg)

**Comment by henkelma** [comment]
![IMG_0253](https://user-images.githubusercontent.com/44764104/57522881-9efde600-7323-11e9-86bd-e76d27ad0f7c.jpeg)

**Comment by holzschu** [comment]
If I type the name of a file at the end of the path in "save as", it saves that file (that is one of the bugs fixed between 1.0 and 1.2). The directory path itself does exist, but the "%20" inside was converted once too many in 1.0.
I am more puzzled about these grayed-out files. Looking closely at your pictures, it seems they were created by Carnets, and you could edit them after creation, but iOS does not recognize them as Jupyter Notebooks, and thus denies Carnets the right to open them, despite the ".ipynb" suffix.
From the Files App, can you open these files?
The most likely explanation is that another App has registered the ".ipynb" suffix as a document type that is subtly different from the definition in Carnets. My goal is now to find which one and how.

**Comment by holzschu** [comment]
Here is, I think, a similar issue (with a different app): https://forums.sketchup.com/t/files-greyed-out/70601

**Comment by Pandaski** [comment]
Same issue here and the reason you supposed is true. The “.ipynb” suffix has been registered by another app “Notebook Master for Jupyter https://itunes.apple.com/cn/app/notebook-master-for-jupyter/id1326802832?mt=8”. After removed that app the notebook is visible in Carnets.

**Comment by sDC2** [comment]
I have a very similar issue: it is not possible for me to add any files from any of my icloud folders into the carnets folder. It is not even possible to copy the "png" files from the welcome folder to a new subfolder that is created in carnets "root".

**Comment by holzschu** [comment]
Hi,
Thank you for pointing this out. I can reproduce the issue. It seems you can copy .ipynb files but not other files.
I think I can have a fix for that rapidly.

**Comment by sDC2** [comment]
Thanks Nicolas for your quick investigation.

Let's hope the fix works, because the addition of screenshots to notebooks is a very helpful feature if it works. Now unfortunately, it is not possible to add images to the Carnets folder.

By the way, how does the "add image"
Option work anyway? The resulting Markdown doesn't provide any clue as to where the picture added is saved - if not transferred to the Carnets folder.

Kind regards,
Sander

On 3 Jan 2020, at 22:54, Nicolas Holzschuch <notifications@github.com> wrote:

﻿

Hi,
Thank you for pointing this out. I can reproduce the issue. It seems you can copy .ipynb files but not other files.
I think I can have a fix for that rapidly.

**Comment by holzschu** [comment]
To add an image to a notebook, I use the markdown syntax: 
```
![title](img/picture.png)
```
The path is relative to the current notebook directory, but it can also be absolute, or an URL.

**Comment by holzschu** [comment]
@sDC2 the issue you reported has been fixed in version 1.3.7, which is released to the AppStore.

**Comment by sDC2** [comment]
Great work! Thanks for the quick solution

On 5 Jan 2020, at 11:58, Nicolas Holzschuch <notifications@github.com> wrote:

﻿

@sDC2<https://apc01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FsDC2&data=02%7C01%7C%7C3de063ea3d094c20317608d791bd6390%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637138114911975867&sdata=vd60QRbDulmj5NZdmI63z5LQLeXNirLtYVmT6mDPCIw%3D&reserved=0> the issue you reported has been fixed in version 1.3.7, which is released to the AppStore.

**Comment by sDC2** [comment]
Hi Nicholas,

Why is there a Scipy reference addd to Carnets, while Carnets doesn't include this library?

On 5 Jan 2020, at 11:58, Nicolas Holzschuch <notifications@github.com> wrote:

﻿

@sDC2<https://apc01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FsDC2&data=02%7C01%7C%7C3de063ea3d094c20317608d791bd6390%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637138114911975867&sdata=vd60QRbDulmj5NZdmI63z5LQLeXNirLtYVmT6mDPCIw%3D&reserved=0> the issue you reported has been fixed in version 1.3.7, which is released to the AppStore.

**Comment by holzschu** [comment]
Hi, 
You mean, in the Help menu? You are right, this is strange. I'll remove it.

**Comment by sDC2** [comment]
Indeed, although adding scipy to the standard modules could be a solution too... The author of Pyto somehow managed to do this...

On 13 Jan 2020, at 23:53, Nicolas Holzschuch <notifications@github.com> wrote:

﻿

Hi,
You mean, in the Help menu? You are right, this is strange. I'll remove it.

**Comment by sDC2** [comment]
Hi Nicolas,

1) I'm trying to download a notebook in pdf format. Unfortunately this generate an error that says pandoc isn't available. Pip instal pandoc does not resolve this. When I check pandoc is installed, pip show shows the module, help("pandoc") throws an error.

[cid:3F2EACAD-448E-4072-8198-8F0144743E08-L0-001]
2) When selecting "download" where does the file get saved? download as text, opens a new window but this window doesn't allow "open in" or "save to files".

thanks for your patience.

kind regards,
sander

**Comment by holzschu** [comment]
Hi, 
pandoc requires a haskell compiler for iOS, which makes it even harder than the usual packages. It is on the todo list, but it will take some time. 

To convert notebooks from ipynb format to other formats, you have `!jupyter-nbconvert notebook.ipynb` which converts to HTML, and jupytext which converts to Markdown (`!pip install jupytext` first).

The "download" behaviour is not useful. I'm going to change it so users can actually save the file in a good place.

**Comment by holzschu** [comment]
Hi again, 
the latest build on TestFlight has a working "Download as" menu. It doesn't allow to download as PDF, but you can download as LaTeX, then convert this LaTeX to PDF using another app, including my other app, [a-Shell.](https://holzschu.github.io/a-Shell_iOS/).
